src/components/WalletUnlock/DeterministicAccountList.tsx

```diff
@@ -88,87 +80,120 @@ const SDownloader = styled(Downloader)`
   }
 `;
 
+export const filterZeroBalanceAccounts = (accounts: DWAccountDisplay[]) =>
+  accounts.filter(({ balance }) => balance?.isZero());
+
 interface DeterministicAccountListProps {
   finishedAccounts: DWAccountDisplay[];
   asset: ExtendedAsset;
   isComplete: boolean;
   network: Network;
-  freshAddressIndex: number;
-  generateFreshAddress(): void;
+  displayEmptyAddresses: boolean;
+  selectedDPath: DPath;
+  handleScanMoreAddresses(dpath: ExtendedDPath): void;
   onUnlock(param: any): void;
   handleUpdate(asset: ExtendedAsset): void;
 }
 
-interface ISelectedAccount {
-  address: TAddress;
-  derivationPath: string;
-}
+const tableBannerText = (emptySelectedAccountsLength: number, maxEmptyAddresses: number) => (
+  <Trans
+    id="DETERMINISTIC_SCANNING_EMPTY_ADDR"
+    variables={{
+      $count: () => emptySelectedAccountsLength,
+      $total: () => maxEmptyAddresses
+    }}
+  />
+);
 
 export default function DeterministicAccountList({
   finishedAccounts,
   asset,
   isComplete,
-  onUnlock,
   network,
-  freshAddressIndex,
-  generateFreshAddress,
+  displayEmptyAddresses,
+  selectedDPath,
+  handleScanMoreAddresses,
+  onUnlock,
   handleUpdate
 }: DeterministicAccountListProps) {
+  const MAX_EMPTY_ADDRESSES = 5;
   const { isMobile } = useScreenSize();
 
-  const [selectedAccounts, setSelectedAccounts] = useState([] as ISelectedAccount[]);
-
-  const accountsToUse = uniqBy(prop('address'), finishedAccounts).filter(
-    ({ isFreshAddress, balance }) => (balance && !balance.isZero()) || isFreshAddress
-  );
+  const [tableAccounts, setTableAccounts] = useState({} as ITableAccounts);
 
+  const accountsToUse = uniqBy(prop('address'), finishedAccounts);
+  // setTableAccounts to be accountsToUse on update with isDefault set if it isn't already set and
   useEffect(() => {
-    const selected = uniqBy(
-      prop('address'),
-      accountsToUse.map(({ address, pathItem }) => ({
-        address,
-        derivationPath: pathItem.path
-      })) as ISelectedAccount[]
-    );
-    setSelectedAccounts(selected);
-  }, [accountsToUse.length]);
+    if (accountsToUse.length === 0 && Object.keys(tableAccounts).length !== 0) {
+      setTableAccounts({} as ITableAccounts);
+    } else {
+      const tableAccs = accountsToUse.reduce((acc, idx) => {
+        acc[idx.address] = tableAccounts[idx.address] || {
+          ...idx,
+          isDefaultConfig: true,
```

Comment:
What does this param mean? Perhaps needs an explainer

Reply:
removed